crates/ef-testing-proc-macro/src/content_reader.rs

```diff
@@ -0,0 +1,54 @@
+use std::collections::BTreeMap;
+
+use serde_json::Value;
+
+use crate::{dir_reader::PathWrapper, utils::blockchain_tests_to_general_state_tests_path};
+
+pub struct ContentReader;
+
+impl ContentReader {
+    pub fn secret_key(path: PathWrapper) -> Result<Value, eyre::Error> {
+        let path = blockchain_tests_to_general_state_tests_path(path);
+        let content = path.read_file_to_string()?;
+
+        let cases: BTreeMap<String, Value> = serde_json::from_str(&content)?;
+        let case = cases
+            .into_values()
+            .collect::<Vec<_>>()
+            .first()
+            .ok_or_else(|| eyre::eyre!("No case found"))?
+            .clone();
+
+        Ok(case
+            .get("transaction")
+            .ok_or_else(|| eyre::eyre!("No transaction found"))?
+            .get("secretKey")
+            .ok_or_else(|| eyre::eyre!("No secret key found"))?
+            .clone())
+    }
+
+    pub fn pre_state(test_case: &Value) -> Result<Value, eyre::Error> {
+        Ok(serde_json::from_value(
+            test_case
+                .get("pre")
+                .ok_or_else(|| eyre::eyre!("No pre state found"))?
+                .clone(),
+        )?)
+    }
+
+    pub fn post_state(test_case: &Value) -> Result<Value, eyre::Error> {
+        Ok(test_case
+            .get("postState")
+            .ok_or_else(|| eyre::eyre!("No post state found"))?
+            .clone())
+    }
+
+    pub fn block(test_case: &Value) -> Result<Value, eyre::Error> {
+        Ok(test_case
+            .get("blocks")
+            .and_then(|blocks| blocks.as_array().map(|blocks| blocks.first()))
```

Comment:
are we sure there's only ever one block? 
will this change? we might have to add a comment to follow up

Reply:
PS: 
What should we prefer here between conciseness and Rust mastery or readability?

```rust
pub fn block(test_case: &Value) -> Result<Value, eyre::Error> {
    // Attempt to get the "blocks" value
    let blocks = test_case
        .get("blocks")
        .ok_or_else(|| eyre::eyre!("Key 'blocks' not found"))?;
    
    // Ensure it's an array
    let blocks_array = blocks
        .as_array()
        .ok_or_else(|| eyre::eyre!("'blocks' is not an array"))?;
    
    // Get the first block
    let first_block = blocks_array
        .first()
        .ok_or_else(|| eyre::eyre!("'blocks' array is empty"))?;

    // Return a clone of the block
    Ok(first_block.clone())
}
```

Reply:
I'm not sure there will always be one block, I have not seen a situation with more than one block, we can update if that happens when debugging the tests.

I tried just to be concise for this function as it doesn't really perform anything interesting, just reads the value from a json, but can update to this if you want.

Reply:
I think it's fine then